File: LeetCode/Palindrome_Number/Palindrome_Number.cpp
// Time: O(n) where n is equal to the number of digits in x
// Space: O(1)

class Solution {
public:
    bool isPalindrome(int x) {
        if (x < 0) {
            return false;
        }
        
        long original = x;
        long reversed = 0;
        while (x != 0) {
            int remainder = x % 10;
            reversed = reversed * 10 + remainder;
            x /= 10;
        }
        return reversed == original;
    }
};

File: LeetCode/Palindrome_Number/Palindrome_Number.py
# Time: O(n) where n is the number of digits in x
# Space: O(1)

class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False

        original = x
        reverse = 0

        while x != 0:
            remainder = x % 10
            reverse = reverse * 10 + remainder
            x //= 10

        return reverse == original
